src/power_grid_model/_utils.py

```diff
@@ -284,3 +287,40 @@ def convert_single_dataset_to_python_single_dataset(data: SingleDataset) -> Sing
         ]
         for component, objects in data.items()
     }
+
+
+def copy_output_to_columnar_dataset(
+    output_data: Dataset,
+    output_component_types: ComponentAttributeMapping,
+    output_type: OutputType,
+    available_components: list[ComponentType],
+) -> Dataset:
+    """Temporary function to copy row based dataset to a column based dataset as per output_component_types.
```

Comment:
Since the new return type in `Dataset` is now somewhat like a `BatchArray` but not really, I think we could augment the definition of `Dataset` in data_types.py to differentiate the `ColumnarArray` from `BatchArray`, which in reality are concepts of two categories. In the end, I think there is little difference to the result recipient and the changes I proposed is merely for semantics.